### ISSUE TITLE
array-bracket-spacing: No linting errors for some arrays that contain line breaks

### ISSUE THREAD

**Issue by nfriend** [eslint/eslint#5768]
**What version of ESLint are you using?**
2.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
Here's my `.eslintrc`:

```javascript
module.exports = {
    "rules": {
        "array-bracket-spacing": [
            "error",
            "never"
        ]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```javascript
var arr = ['foo',
  'bar'
];
```

**What did you expect to happen?**
I expected this code to cause a linting error.  This code is taken from [the documentation](http://eslint.org/docs/rules/array-bracket-spacing), where it is currently the fifth example of "patterns that are considered problems" under the "Never" option heading.

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
PS C:\temp> eslint .\test.js
PS C:\temp>
```

**Comment by mysticatea** [comment]
Thank you for this issue.

Surely, the document says that code is a problem, but the implementation doesn't report it.
But I'm not sure which is wrong.

**Comment by alberto** [comment]
It's a doc issue.
